dashboards-observability/public/components/visualizations/charts/box_plot/box_plot.tsx

```diff
@@ -0,0 +1,190 @@
+    }
+    return prev.map(
+      (item: string | number, index: number) => `${item},<br>${data[cur.name][index]}`
+    );
+  }, []);
+
+  box = valueSeries.map((field: ConfigListEntry, index: number) => {
+    const selectedColor = getSelectedColorTheme(field, index);
+    const axis = isVertical ? 'y' : 'x';
+    const alternateAxis = isVertical ? 'x' : 'y';
+    return {
+      [axis]: data[field.name],
+      ...(boxMode === 'group' && {
+        [alternateAxis]: dimensionData,
+        orientation: boxOrientation,
+      }),
+      boxpoints: 'all',
+      jitter: jitter,
+      pointpos: PointPosition,
+      type: 'box',
+      line: {
+        color: selectedColor,
+      },
+      fillcolor: hexToRgb(selectedColor, fillOpacity),
+      marker: {
+        size: markerSize,
+        color: selectedColor,
+        opacity: fillOpacity,
+        line: {
+          color: selectedColor,
+          width: 4,
+        },
+      },
+      name: field.name,
+      boxmean: true,
+    };
+  });
+
+  // then use the LONG_CHART_COLOR for all the box in the chart
+  const plotlyColorway =
+    data[fields[lastIndex].name].length < 16 ? PLOTLY_COLOR : [LONG_CHART_COLOR];
+  const mergedLayout = {
+    colorway: plotlyColorway,
+    ...layout,
+    ...(layoutConfig.layout && layoutConfig.layout),
+    title: dataConfig?.panelOptions?.title || layoutConfig.layout?.title || '',
+    boxmode: boxMode,
+    xaxis: {
+      ...(isVertical && { tickangle: tickAngle }),
+      automargin: true,
+      tickfont: {
+        ...(labelSize && {
+          size: labelSize,
+        }),
+      },
+    },
+    yaxis: {
+      ...(!isVertical && { tickangle: tickAngle }),
+      automargin: true,
+      tickfont: {
+        ...(labelSize && {
+          size: labelSize,
+        }),
+      },
+    },
+    boxgap: boxGap,
+    legend: {
+      ...layout.legend,
+      orientation: legendPosition,
+      ...(legendSize && {
+        font: {
+          size: legendSize,
+        },
+      }),
+    },
+    showlegend: showLegend,
+  };
+  if (availabilityConfig.level) {
+    const thresholdTraces = {
+      x: [],
+      y: [],
+      mode: 'text',
+      text: [],
+    };
+    const levels = availabilityConfig.level ? availabilityConfig.level : [];
+
+    const mapToLine = (list: ThresholdUnitType[] | AvailabilityUnitType[], lineStyle: any) => {
+      return list.map((thr: ThresholdUnitType) => {
+        thresholdTraces.x.push(
+          data[!isEmpty(xaxis) ? xaxis[xaxis.length - 1]?.label : fields[lastIndex].name][0]
+        );
+        thresholdTraces.y.push(thr.value * (1 + 0.06));
+        thresholdTraces.text.push(thr.name);
+        return {
+          type: 'box',
+          x0: data[!isEmpty(xaxis) ? xaxis[0]?.label : fields[lastIndex].name][0],
+          y0: thr.value,
+          x1: last(data[!isEmpty(xaxis) ? xaxis[0]?.label : fields[lastIndex].name]),
+          y1: thr.value,
+          name: thr.name || '',
+          opacity: 0.7,
```

Comment:
check significance of this once.

Reply:
no threshold in chart config of box plot. But this block is for levels from availabilityConfig, so i haven't removed this one @abasatwar